src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala

```diff
@@ -1024,26 +1023,42 @@ trait BronzeTransforms extends SparkSessionWrapper {
     // This captures clusters that have not been edited/restarted since the last run with
     // log confs as they will not be in the audit logs
     val latestSnapW = Window.partitionBy('organization_id, 'cluster_id).orderBy('Pipeline_SnapTS.desc)
-    val newLogDirsNotIdentifiedInAudit = clusterSnapshot
-      .join(incrementalClusterIDs, Seq("cluster_id"))
-      .withColumn("snapRnk", rank.over(latestSnapW))
-      .filter('snapRnk === 1)
-      .withColumn("cluster_log_conf",
-        fetchClusterLogConfiguration(cloudProvider,coalesce($"cluster_log_conf.dbfs.destination", $"cluster_log_conf.s3.destination"),
-          isMultiWorkSpaceDeployment, organisationId))
-      .filter('cluster_id.isNotNull && 'cluster_log_conf.isNotNull)
-      .select('cluster_id, 'cluster_log_conf)
+
+    val newLogDirsNotIdentifiedInAudit = {
+      if(clusterSnapshot.isEmpty){
+        spark.emptyDataFrame
+      }else{
+        clusterSnapshot
+          .join(incrementalClusterIDs, Seq("cluster_id"))
+          .withColumn("snapRnk", rank.over(latestSnapW))
```

Comment:
@sriram251-code we should use row_number instead of rank, just we avoid a condition where we have multiple rows having rank = 1

Reply:
Point well taken but the code was previously present i have just added some exception handling. Moving to row_number might need some regression testing that we can achieve after our validation module.

Reply:
okay